Rasa_Bot/actions/helper.py

```diff
@@ -264,6 +263,109 @@ def get_random_activities(avoid_activity_id: int, number_of_activities: int
+                question_id: the question_id for which the answers should be retrieved
+
+            Returns:
+                    All the answers that the user has given for the given question
+
+    """
+    session = get_db_session(db_url=DATABASE_URL)
+
+    closed_answers = (
+        session.query(
+            DialogClosedAnswers
+        )
+        .join(ClosedAnswers)
+        .filter(
+            DialogClosedAnswers.users_nicedayuid == user_id,
+            ClosedAnswers.question_id == question_id
+        )
+        .all()
+    )
+
+    return closed_answers
+
+def get_all_closed_answers(question_id):
+    """
+       Get all the possible closed answers associated with a given question id.
+        Args:
+                question_id: the question_id for which the answers should be retrieved
+
+            Returns:
+                    All the possible answers to the question specified
+
+    """
+    session = get_db_session(db_url=DATABASE_URL)
+
+    closed_answers = (
+        session.query(
+            ClosedAnswers
+        )
+        .filter(
+            ClosedAnswers.question_id == question_id
+        )
+        .all()
+    )
+
+    return closed_answers
+
+def get_open_answers(user_id, question_id):
+    """
+       Get the open answer responses associated with the given user and question.
+        Args:
+                user_id: the user_id of the user to retrieve the answers for
+                question_id: the question_id for which the answers should be retrieved
+
+            Returns:
+                    The open answers that the user has given for the question
+
+    """
+    session = get_db_session(db_url=DATABASE_URL)
+
+    closed_answers = (
+        session.query(
+            DialogOpenAnswers
+        )
+        .filter(
+            DialogOpenAnswers.users_nicedayuid == user_id,
+            DialogOpenAnswers.question_id == question_id
+        )
+        .all()
+    )
+
+    return closed_answers
+
+def count_answers(answers, closed_answer_options):
+    """
+       Count up the closed_answer responses for each answer option.
+        Args:
+                answers: the answers of the user, retreived from the database
+                closed_answer_options: the answer options for the given question
+
+            Returns:
+                    An array of values, specifying the amount of occurences of an answer
+                    for the corresponding option, aligning with the index of the option.
+                    Example... [2, 1, 3] for ["At the gym", "At home", "Outside"]
+
+    """
+    result = []
+    for closed_answer_option in closed_answer_options:
+        amount_of_answers = 0
+        closed_answer = closed_answer_option
+
+        for answer in answers:
+            if answer.closed_answers_id == closed_answer.closed_answers_id:
+                amount_of_answers += 1
+
+        result.append(amount_of_answers)
```

Comment:
You can use nested comprehension lists to do this in a more performant way.

```suggestion
       result = [
        len(
            [answer for answer in answers if answer.closed_answers_id == closed_answer.closed_answers_id]
        ) for
              closed_answer in closed_answer_options
    ]
```